.gitignore

```diff
@@ -1,2 +1,334 @@
-bin/
-obj/
+## Ignore Visual Studio temporary files, build results, and
```

Comment:
Any specific reason for all these .gitignore changes?

Reply:
I did this because my IDE creates a lot of redundant files (especially `.vs` folder) that I didn't want to commit those files by mistake.

It's not suitable for this PR, so this commit is going to be reverted. I would to create a separate issue if it's important for the future. 

I took this file from https://github.com/github/gitignore/blob/main/VisualStudio.gitignore

![without_big_gitignore](https://github.com/pgvector/pgvector-dotnet/assets/9298580/3c01e42a-d618-43ec-90c4-b800de6a6b20)